src/resource_clients/dataset.ts

```diff
@@ -156,29 +123,75 @@ class DatasetClient extends ResourceClient {
         });
     }
 
-    /**
-     * @param response
-     * @return {PaginationList}
-     * @private
-     */
-    _createPaginationList(response) {
+    private _createPaginationList<D>(response: AxiosResponse<D[]>): PaginatedList<D> {
         return {
             items: response.data,
             total: Number(response.headers['x-apify-pagination-total']),
             offset: Number(response.headers['x-apify-pagination-offset']),
             count: response.data.length, // because x-apify-pagination-count returns invalid values when hidden/empty items are skipped
             limit: Number(response.headers['x-apify-pagination-limit']), // API returns 999999999999 when no limit is used
+            // TODO: will the API return this
+            desc: false,
         };
     }
 }
 
-module.exports = DatasetClient;
+type MaybeArray<T> = T | T[];
 
-/**
- * @typedef {object} PaginationList
- * @property {object[]} items - List of returned objects
- * @property {number} total - Total number of objects
- * @property {number} offset - Number of objects that were skipped
- * @property {number} count - Number of returned objects
- * @property {number} limit - Requested limit
- */
+export interface Dataset {
+    id: string;
+    name: string;
+    userId: string;
+    createdAt: string;
+    modifiedAt: string;
+    accessedAt: string;
+    itemCount: number;
+    cleanItemCount: number;
+    // TODO: Are these actually always just empty objects?
+    actId?: Record<string, unknown> | null;
+    actRunId?: Record<string, unknown> | null;
+    fields?: string[];
+    // TODO: the api seems to return a stats field, that isn't documented
```

Comment:
cc @mtrunkat, can you please shed some light on this, or suggest someone who can help?

(basically all the questions marked with TODO in this PR)